app/services/create_location_decorator.rb

```diff
@@ -0,0 +1,13 @@
+class CreateLocationDecorator
```

Comment:
The problem as I see it is that in order to present a `Location` to the view we now have to decorate it along with **three** other collaborating objects.

This service is just a factory method but I do like having just one thing to pass all the parameters to and getting back the object for the view to present. Moving the logic out of the controller I think is a good move especially seeing how we have agreed to stick with Rubocop's guidelines when it comes to method length and complexity.

Having one object construct a decorated object is not inherently bad I think; we have a measure of complexity to deal with and the priority should be to make that as easy to follow as possible for our future selves.

If we think that having an object to coordinate things is a good idea then perhaps we could go a step further and have it in charge of all interactions within the controller action, say something like `FindAndDecorateLocation`.

I see little value in having something that returns a list of values only then for those values to be piped straight into a constructor.

Probably best to discuss this face-to-face tomorrow!